.eslintrc.js

```diff
@@ -0,0 +1,130 @@
+/* eslint-disable-next-line no-undef */
+module.exports = {
+    parser: '@typescript-eslint/parser',
+    parserOptions: {
+        ecmaVersion: 2022,
+        sourceType: 'module',
+        extraFileExtensions: [
+            '.json'
+        ]
+    },
+    env: {
+        node: true,
+        es6: true,
+        jest: true
+    },
+    plugins: [
+        '@typescript-eslint',
+        'eslint-plugin-json'
+    ],
+    extends: [
+        'eslint:recommended'
+    ],
+    root: true,
+
+    globals: {},
+
+    overrides: [
+        {
+            files: [
+                '*.ts'
+            ],
+            extends: [
+                'plugin:@typescript-eslint/recommended'
+            ],
+            parserOptions: {
+                project: [
+                    './tsconfig.json'
+                ]
+            }
+        }
+    ],
+
+    // 'off' or 0 - turn the rule off
+    // 'warn' or 1 - turn the rule on as a warning (doesn’t affect exit code)
+    // 'error' or 2 - turn the rule on as an error (exit code will be 1)
+    rules: {
+        'block-scoped-var': 'error',
+        complexity: ['error', { max: 40 }],
+        'consistent-return': 'error',
+        curly: 'error',
+        'dot-location': ['error', 'property'],
+        'dot-notation': ['error', { allowPattern: '^[a-z]+(_[a-z]+)+$' }],
+        'no-alert': 'error',
+        'no-multi-spaces': 'error',
+        'no-redeclare': 'error',
+        'no-unused-expressions': ['error', { allowShortCircuit: true, allowTernary: true }],
+        'vars-on-top': 'off',
+        yoda: ['error', 'never', { exceptRange: true }],
+        'no-console': 'warn', // allow debugging
+
+        // Stylistic Issues and Opinions
+        'arrow-body-style': 'error',
+        'array-bracket-spacing': ['error', 'never'],
+        'arrow-parens': ['error', 'as-needed'],
+        'arrow-spacing': 'error',
+        'block-spacing': ['error', 'always'],
+        'brace-style': ['error', '1tbs', { allowSingleLine: true }],
+        camelcase: 'error',
+        'comma-dangle': ['error', 'never'],
+        'comma-spacing': ['error', { before: false, after: true }],
+        'comma-style': 'error',
+        'computed-property-spacing': ['error', 'never'],
+        'function-paren-newline': ['error', 'consistent'],
+        indent: [
+            'error',
```

Comment:
I noticed in VSCode that I needed to enable eslint as a formatter or my editor was very unhappy with this